Packages/eu.netherlands3d.indicators/Runtime/Scripts/DossierSO.cs

```diff
@@ -76,18 +78,46 @@ public string ApiKey
                 
                 // since we do not support multiple frames at the moment, we cheat and always load the first
                 var firstFrame = value.Value.frames.First();
-                var firstFrameMap = firstFrame.map;
-                if (!string.IsNullOrEmpty(apiKey))
-                {
-                    var uriBuilder = new UriBuilder(firstFrameMap);
-                    uriBuilder.Query = string.IsNullOrEmpty(uriBuilder.Query) 
-                        ? $"code={apiKey}" 
-                        : string.Concat(uriBuilder.Query, $"&code={apiKey}");
-                    firstFrameMap = uriBuilder.Uri;
-                }
+                var firstFrameMapUrl = AppendDossierCodeToURL(firstFrame.map);
+                
+                // We parse the frame map data
+                onLoadMapOverlayFrame.Invoke(firstFrameMapUrl);
+            }
+        }
 
-                onLoadMapOverlayFrame.Invoke(firstFrameMap);
+        public Uri AppendDossierCodeToURL(Uri url)
```

Comment:
Name is not entirely accurate, the Uri has the API Key added to it and not the DossierCode

Reply:
Fixed